gnovm/pkg/gnolang/go2gno_test.go

```diff
@@ -25,3 +28,303 @@ func main(){
 	fmt.Printf("AST:\n%#v\n\n", n)
 	fmt.Printf("AST.String():\n%s\n", n.String())
 }
+
+type mockPackageGetter []*std.MemPackage
+
+func (mi mockPackageGetter) GetMemPackage(path string) *std.MemPackage {
+	for _, pkg := range mi {
+		if pkg.Path == path {
+			return pkg
+		}
+	}
+	return nil
+}
+
+type mockPackageGetterCounts struct {
+	mockPackageGetter
+	counts map[string]int
+}
+
+func (mpg mockPackageGetterCounts) GetMemPackage(path string) *std.MemPackage {
+	mpg.counts[path]++
+	return mpg.mockPackageGetter.GetMemPackage(path)
+}
+
+func TestTypeCheckMemPackage(t *testing.T) {
```

Comment:
Is it even possible to run this test / the cases in parallel?

Reply:
fair point; added parallel tests...